nexus/Cargo.toml

```diff
@@ -121,6 +121,7 @@ itertools = "0.10.5"
 nexus-test-utils-macros = { path = "test-utils-macros" }
 nexus-test-utils = { path = "test-utils" }
 omicron-test-utils = { path = "../test-utils" }
+omicron-sled-agent = { path = "../sled-agent" }
```

Comment:
Yeah, this is what the gist of the PR description was saying -- these tests are co-located with Nexus, so I feel "more okay" doing things like poking at saga internals, but they're basically integration tests, not unit tests. Functionally, they are more aligned with the tests in `nexus/tests/integration_tests`, but they happen to poke at more internal things -- so, like those tests, they want to depend on `omicron-sled-agent` to poke at the `sim` interfaces.

If it helps, this is a `dev-dependency`, not a `dependency`, but maybe it's worthwhile splitting the "real" and "simulated" sled agents into distinct crates?

Reply:
I defer to you on that.  I don't spend enough time in those to know if it would be useful.  I don't think this change here really changes much since I think we must already have been depending on it via nexus-test-utils.

Reply:
It was a transitive dependency before, but in this test, we actually directly use the `omicron-sled-agent` crate from the Nexus to access the "sim" interfaces. This is used to access the simulated object, and ensure that no "simulated crucible regions" exist within the test.

If we didn't need to access that interface, I don't think this dependency would be necessary.

Reply:
To be specific: See the `sled_agent.get_crucible_dataset(...)` invocation in `no_regions_ensured`